script/validate-data/index.ts

```diff
@@ -104,9 +105,19 @@ async function checkWorkflow(workflowPath: string, propertiesPath: string, allow
       }
       
     }
-    if (!workflowPath.endsWith("blank.yml") && (!properties.categories || 
-      !properties.categories.some(category => allowed_categories.some(ac => ac.toLowerCase() == category.toLowerCase())))) {
-      workflowErrors.errors.push(`Workflow does not contain at least one allowed category - ${allowed_categories}`)
+    var folderName = dirname(workflowPath)
+    var folder_category = folder_category_map.find( folder_category => folder_category["name"] == folderName)["category"]
+    if (!workflowPath.endsWith("blank.yml") && ((!properties.categories || properties.categories.length == 0 )|| 
+      properties.categories[0].toLowerCase() !== folder_category.toLowerCase())) {
+      if(!properties.categories || properties.categories.length == 0) {
+        workflowErrors.errors.push(`Workflow categories cannot be null or empty`)
+      } else {
+        workflowErrors.errors.push(`The first category in properties.json categories must be "${folder_category}" for ${basename(folderName)} folder workflow.`)
+      }
+    }
```

Comment:
This can be simplified. Handle `!properties.categories || properties.categories.length == 0` separately and `properties.categories[0].toLowerCase() !== folder_category.toLowerCase()` separately.

Reply:
done